src/util/map/WrappedSetMultiMap.ts

```diff
@@ -0,0 +1,151 @@
+import type { SetMultiMap } from './SetMultiMap';
+
+/**
+ * A {@link SetMultiMap} that uses an input map to store its data.
+ * The map that is provided to the constructor needs to be empty,
+ * and not be interacted with anymore.
```

Comment:
Why would I want to use this though?
Why not just initialize the map ourselves, and expose it, if it needs to be empty anyway?

But why is the map exposed in the first place? I don't understand the purpose of this class.

Reply:
> Why would I want to use this though?

If you are a bad developer 😛.

The reason is that this allows us to wrap around different Map implementations. So this can currently use this with a `Map` or a `HashMap`. And more if new Maps get added. So in short modularity. I also don't completely like that an existing object gets passed in. I actually thought about taking a constructor as input parameter ... but then I forgot to look into that 😄 .

Reply:
Yeah let's maybe not allow that then, for now. Would do as needed.

Reply:
> Yeah let's maybe not allow that then, for now. Would do as needed.

So your suggestion is to have a `HashMultiMapSet` that extends `HashMap` instead and also take a hash function as input? I wanted to enable the possibility of also having a `SetMultiMap` that uses a `Map` as there might be use cases for that (although currently there are none).